GLSP-1022: Update dependencies

tortmayr opened the pull request:
- Update to latest next version of glsp dev packages and sprotty
  - This also requires an update to inversify 6
- Update to TS 6

## Sprotty update
- Move sprotty-reexport.ts into `glsp-sprotty´ directory  and provide `~glsp-sprotty` path alias to ensure that client-internal we also only reuse the augmented `glsp-sprotty` reexport with having to use relative paths.
- Counteract the breaking sprotty naming changes by rexporting them as the old types (SModelElementImpl-> SModelElement)
- Adapt sprotty an glsp-protocol imports of `@eclipse-glsp/client` to consistently use `~glsp-sprotty` instead This reexport is a glsp-ified version of the sprotty main index that augments some modules of sprotty (actions,types) and replaces some modules with a specific glsp version.

Part of https://github.com/eclipse-glsp/glsp/pull/1022

Note: This change does not include any adaptions regarding SModel vs GModel (https://github.com/eclipse-glsp/glsp/issues/865). This can be tackled in a follow-up PR.

martin-fleck-at commented on packages/client/src/features/context-menu/delete-element-context-menu.ts:
```diff
@@ -13,11 +13,10 @@
+import { DeleteElementOperation, IContextMenuItemProvider, MenuItem, Point, SModelRoot } from '~glsp-sprotty';
 import { EditorContextService, EditorContextServiceProvider } from '../../base/editor-context-service';
-import { TYPES } from '../../base/types';
+import { TYPES } from '../../glsp-sprotty/types';
```
nit: should be added to the imports of `~glsp-sprotty`

martin-fleck-at commented on packages/client/src/features/command-palette/server-command-palette-provider.ts:
```diff
@@ -13,12 +13,19 @@
 import { GLSPActionDispatcher } from '../../base/action-dispatcher';
 import { EditorContextService } from '../../base/editor-context-service';
-import { TYPES } from '../../base/types';
+import { TYPES } from '../../glsp-sprotty/types';
```
nit: should be added to the imports of `~glsp-sprotty`

martin-fleck-at commented on packages/client/src/features/command-palette/di.config.ts:
```diff
@@ -13,11 +13,10 @@
+import { CommandPalette, CommandPaletteActionProviderRegistry, bindAsService } from '~glsp-sprotty';
 import '../../../css/command-palette.css';
-import { TYPES } from '../../base/types';
+import { TYPES } from '../../glsp-sprotty/types';
```
nit: should be added to the imports of `~glsp-sprotty`

martin-fleck-at commented on packages/client/src/features/context-menu/di.config.ts:
```diff
@@ -13,10 +13,9 @@
-import { ContextMenuProviderRegistry, IContextMenuService } from 'sprotty';
-import { TYPES } from '../../base/types';
+import { ContextMenuProviderRegistry, IContextMenuService, bindAsService } from '~glsp-sprotty';
+import { TYPES } from '../../glsp-sprotty/types';
```
nit: should be added to the imports of `~glsp-sprotty`
(applies to all subsequent imports as well, I just don't want to push too many comments for the same thing ;-))

martin-fleck-at commented on packages/client/src/features/tool-feedback/creation-tool-feedback.ts:
```diff
@@ -13,27 +13,30 @@
  ********************************************************************************/
-import { Action, Bounds, Point, SEdgeSchema } from '@eclipse-glsp/protocol';
 import { inject, injectable } from 'inversify';
+import { MoveAction } from 'sprotty-protocol/lib/actions';
```
nit: Should be imported from `~glsp-sprotty`.

martin-fleck-at commented on packages/client/src/features/tool-feedback/edge-edit-tool-feedback.ts:
```diff
@@ -13,31 +13,34 @@
-import { Action, Bounds, hasStringProp, Point } from '@eclipse-glsp/protocol';
 import { inject, injectable } from 'inversify';
 import { VNode } from 'snabbdom';
+import { MoveAction } from 'sprotty-protocol/lib/actions';
```
nit: Should be imported from `~glsp-sprotty`.

martin-fleck-at commented on packages/client/src/model/update-model-command.ts:
```diff
@@ -0,0 +1,126 @@
+/********************************************************************************
```
This seems to be a duplicate of the `packages/client/src/base/model/update-model-command.ts`. I assume this was copied by mistake?

Nevertheless, I'm wondering whether we should add the action handler and/or update model command from this file to the glsp-sprotty export as well? At least the `FeedbackAwareUpdateModelCommand` seems to be a specialization of the `UpdateModelCommand`. Thoughts?

tortmayr replied:
It is a specialization of the update model command. However, it is only relevant in the glsp context so I don't see a reason to move this into the sprotty export (as it is not replacing any direct sprotty contribution). Also it has dependencies to other GLSP client concepts like the `FeedbackActionDispatcher` and model listeners. We would introduce a dependency circle by moving it into the sprotty export.

tortmayr replied:
I have removed the duplicate class

martin-fleck-at commented on packages/client/src/glsp-sprotty/index.ts:
```diff
@@ -230,12 +260,12 @@ export {
+    ShapedPreRenderedElementImpl as ShapedPreRenderedElement
 } from 'sprotty/lib/lib/model';
 // ------------------ Library ------------------
 export * from 'sprotty/lib/lib/modules';
```
Should we also `export * from 'sprotty/lib/lib/virtualize';`? Seems odd that this is omitted or is it because of the default export?

tortmayr replied:
Sure we can add it. It's not part of the original sprotty main index, but I don't see a reason why we should not reexport those symbols as well.

martin-fleck-at reviewed (changes requested):
Change looks good to me for the most part and I really like the additional layer that we now have between sprotty and GLSP through the complete re-export mechanism! Great work as always! I do have a few comments on which I'd like your input but other than that it is mostly ready to be merged.

tortmayr commented on package.json:
```diff
@@ -27,9 +27,12 @@
   },
   "devDependencies": {
-    "@eclipse-glsp/dev": "1.1.0-next.164cf99.124",
+    "@eclipse-glsp/dev": "next",
```
The version is locked via yarn lock anyways. It will only be bumped if we explicitly upgrade the version via `yarn upgrade:next`.

tortmayr commented:
Thanks for the review Martin. I have updated the imports in question to consistently import for `~glsp-sprotty` and added `export * from 'sprotty/lib/lib/virtualize' to the sprotty reexport.

martin-fleck-at reviewed (approved):
Fantastic, changes look good to me, thank you Tobias!
